src/core/core_utils.js

```diff
@@ -205,7 +205,13 @@ function escapePDFName(str) {
   let start = 0;
   for (let i = 0, ii = str.length; i < ii; i++) {
     const char = str.charCodeAt(i);
-    if (char < 0x21 || char > 0x7e || char === 0x23) {
+    if (
+      char < 0x21 ||
+      char > 0x7e ||
+      char === 0x23 /* # */ ||
+      char === 0x28 /* ( */ ||
+      char === 0x29 /* ) */
+    ) {
```

Comment:
A Name object should escape ALL possible delemitter and not only these 3. So I would say that these are still missing: `<>[]{}\%`

Reply:
That's correct, I'm fixing that, thx.

Reply:
For our reference, is this mentioned somewhere in the PDF specification so we can verify this patch? The change itself looks good if for all characters a unit test is available, but it would be good to check with the specification too.

Reply:
In PDF 32000-1:2008 it is 7.5.3:

> a) A NUMBER SIGN (23h) (#) in a name shall be written by using its 2-digit hexadecimal code (23), preceded by the NUMBER SIGN.
b ) Any character in a name that is a regular character (other than NUMBER SIGN) shall be written as itself or by using its 2-digit hexadecimal code, preceded by the NUMBER SIGN.
c) Any character that is not a regular character shall be written using its 2-digit hexadecimal code, preceded by the NUMBER SIGN only.

> White space used as part of a name shall always be coded using the 2-digit hexadecimal notation and no white space may intervene between the SOLIDUS and the encoded name.
> Regular characters that are outside the range EXCLAMATION MARK(21h) (!) to TILDE (7Eh) (~) should be written using the hexadecimal notation.

The quoted characters are all delemiter characters (see 7.2.2 Character Set). If you would not escape them a new object will begin ;-)

Reply:
For future reference: it's in section 7.3.5, not 7.5.3, but it does allow us to check this patch against. Thanks!

Reply:
ups, sorry... sure 7.3.5! It was late :-)